lib/podlet.js

```diff
@@ -811,10 +824,50 @@ export default class PodiumPodlet {
      * @returns {string}
      */
     render(incoming, data, ...args) {
+        const markup = this.#useShadowDOM ? this.wrapWithShadowDOM(data) : data;
         if (!incoming.development) {
-            return data;
+            return markup;
         }
-        return this.#view(incoming, data, ...args);
+        return this.#view(incoming, markup, ...args);
+    }
+
+    /**
+     * Function that takes in the podlet content,wraps it in declarative shadow DOM and returns it.
+     * @param {string} data - the podlet content to wrap in declarative shadow DOM as an HTML markup string
+     *
+     * @example
+     * ```js
+     * const content = podlet.wrapWithShadowDOM('<div>content to render</div>');
+     * ```
+     */
+    wrapWithShadowDOM(data) {
+        assert.ok(
+            customElementRegex.test(this.name),
+            'When using the constructor argument "useShadowDOM", podlet.name must conform to custom element naming conventions: https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/define#valid_custom_element_names',
+        );
+
+        const styles = this.cssRoute
+            .filter((css) => css.scope === 'shadow-dom')
+            .map((css) => css.toHTML());
+
+        const scripts = this.jsRoute
+            .filter((js) => js.scope === 'shadow-dom')
+            .map((js) => js.toHTML());
+
+        // Wrap the markup in DSD.
+        return `
+          <${this.name}>
+            <template shadowrootmode="open">
+              ${styles.join('\n')}
+              ${data}
+              ${scripts.join('\n')}
+            </template>
+          </${this.name}>
+          <script>(()=>{function e(d){HTMLTemplateElement.prototype.hasOwnProperty("shadowRootMode")||d.querySelectorAll("template[shadowrootmode]").forEach(o=>{let
+          n=o.getAttribute("shadowrootmode"),s=o.hasAttribute("shadowrootdelegatesfocus"),t=o.parentNode.attachShadow({mode:n,delegatesFocus:s});t.appendChild(o.content),o.remove(),e(t)})}var
+          r;(r=document.currentScript)!=null&&r.previousElementSibling&&e(document.currentScript.previousElementSibling);})();
+          </script>
```

Comment:
Is this needed any more?

Reply:
This is the DSD polyfill, I don't think we can let go of this just yet, but I guess hopefully not too long